realm/realm-library/src/androidTest/java/io/realm/RealmListTest.java

```diff
@@ -717,6 +717,61 @@ public void testClearNotDeleting() {
         testRealm.commitTransaction();
     }
 
+    public void testContains() {
+        Owner owner = testRealm.where(Owner.class).findFirst();
+        Dog dog = owner.getDogs().get(0);
+        assertTrue("Should contain a particular dog.", owner.getDogs().contains(dog));
+    }
+
+    public void testContainsNotManaged() {
+        Owner owner = testRealm.where(Owner.class).findFirst();
+        RealmList<Dog> managedDogs = owner.getDogs();
+        // Create a unmanaged RealmList
+        RealmList<Dog> unmanagedDogs
+                = new RealmList<Dog>(managedDogs.toArray(new Dog[managedDogs.size()]));
+        Dog dog = managedDogs.get(0);
+        assertTrue("Should contain a particular dog", unmanagedDogs.contains(dog));
+    }
+
+    public void testContainsNull() {
+        Owner owner = testRealm.where(Owner.class).findFirst();
+        assertFalse("Should not contain a null item.", owner.getDogs().contains(null));
+    }
+
+    /**
+     * This test requires an additional Realm to test against.
+     */
+    public void testContainsDoesNotContainAnItem() {
+        RealmConfiguration realmConfig = TestHelper.createConfiguration(getContext(), "contains_test.realm");
+        Realm.deleteRealm(realmConfig);
+        Realm testRealmTwo = Realm.getInstance(realmConfig);
+        try {
+            // Set up the test realm
+            testRealmTwo.beginTransaction();
+            Owner owner2 = testRealmTwo.createObject(Owner.class);
+            owner2.setName("Owner");
+            for (int i = 0; i < TEST_OBJECTS; i++) {
+                Dog dog = testRealmTwo.createObject(Dog.class);
+                dog.setName("Dog " + i);
+                owner2.getDogs().add(dog);
+            }
+            testRealmTwo.commitTransaction();
+
+            // Get a dog from the test realm.
+            Dog dog2 = testRealmTwo.where(Owner.class).findFirst().getDogs().get(0);
+
+            // Access the original Realm. Then see if the above dog object is contained. (It shouldn't).
+            Owner owner1 = testRealm.where(Owner.class).findFirst();
+
+            assertFalse("Should not be able to find one object in another Realm via contains",
+                    owner1.getDogs().contains(dog2));
+        } finally {
+            if(testRealmTwo != null && !testRealmTwo.isClosed()) {
+                testRealmTwo.close();
+            }
+        }
+    }
+
```

Comment:
One more test for an Object is in the same realm with the RealmList but not contained :)